app/models/good_job/execution.rb

```diff
@@ -316,10 +316,21 @@ def executable?
       self.class.unscoped.unfinished.owns_advisory_locked.exists?(id: id)
     end
 
-    def active_job
-      ActiveJob::Base.deserialize(active_job_data).tap do |aj|
-        aj.send(:deserialize_arguments_if_needed)
+    # Build an ActiveJob instance and deserialize the arguments, using `#active_job_data`.
+    #
+    # @param ignore_deserialization_errors [Boolean]
+    #   Whether to ignore ActiveJob::DeserializationError when deserializing the arguments.
+    #   This is most useful if you aren't planning to use the arguments directly.
+    def active_job(ignore_deserialization_errors: false)
+      instance = ActiveJob::Base.deserialize(active_job_data)
+
+      begin
+        instance.send(:deserialize_arguments_if_needed)
+      rescue ActiveJob::DeserializationError
+        raise unless ignore_deserialization_errors
       end
+
+      instance
```

Comment:
I deviated slightly from the suggestion on the issue here, I think it makes sense to return an instance of the job even if the arguments couldn't be deserialized (although this could introduce further bugs downstream if you tried to do other things with it....)

Reply:
Yes! In strong agreement 👍🏻

Reply:
Looking a bit closer at that lint error, I realized that the `tap` in my original example was ensuring that the activejob instance is returned regardless of what happens in the rescue

Reply:
🤦 you're right of course - I've reverted to that style.